zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -662,6 +662,20 @@ public void startdata() throws IOException, InterruptedException {
     }
 
     public synchronized void startup() {
+        startupWithServerState(State.RUNNING);
+        notifyAll();
+    }
+
+    public synchronized void startupWithoutServing() {
+        startupWithServerState(State.INITIAL);
+        notifyAll();
+    }
+
+    public synchronized void startServing() {
+        setState(State.RUNNING);
```

Comment:
notifyAll(); here may be needed 
to wakeup the thread sleep on the condition of State.INITIAL.
-------------------------
public void enqueueRequest(Request si)
public void submitRequestNow(Request si)
sleep at State.INITIAL

Reply:
good catch - there is no correctness issue here without notifying because the wait will timeout anyway and next check will catch the correct state, but better to notify even the timeout is pretty short to avoid unnecessary wait.